DatadogLogs/Sources/Logger.swift

```diff
@@ -7,247 +7,183 @@
+            case short
+            /// Prints short representation of log with given prefix.
+            case shortWith(prefix: String)
+        }
 
-    /// Sends a WARN log message.
-    /// - Parameters:
-    ///   - message: the message to be logged
-    ///   - error: the error information (optional)
-    ///   - attributes: a dictionary of attributes (optional) to add for this message. If an attribute with
-    /// the same key already exist in this logger, it will be overridden (only for this message).
-    func warn(_ message: String, error: Error? = nil, attributes: [AttributeKey: AttributeValue]? = nil) {
-        log(level: .warn, message: message, error: error, attributes: attributes)
+        /// The service name  (default value is set to application bundle identifier)
+        public var serviceName: String?
+
+        /// The logger custom name (default value is set to main bundle identifier)
+        public var loggerName: String?
+
+        /// Enriches logs with network connection info.
+        /// This means: reachability status, connection type, mobile carrier name and many more will be added to each log.
+        /// For full list of network info attributes see `NetworkConnectionInfo` and `CarrierInfo`.
+        ///
+        /// `false` by default.
+        public var sendNetworkInfo: Bool
+
+        /// Enables the logs integration with RUM.
+        /// If enabled all the logs will be enriched with the current RUM View information and
+        /// it will be possible to see all the logs sent during a specific View lifespan in the RUM Explorer.
+        ///
+        /// `true` by default.
+        public var bundleWithRUM: Bool
+
+        /// Enables the logs integration with active span API from Tracing.
+        /// If enabled all the logs will be bundled with the `DatadogTracer.shared().activeSpan` trace and
+        /// it will be possible to see all the logs sent during that specific trace.
+        ///
+        /// `true` by default.
+        public var bundleWithTrace: Bool
+
+        /// Enables logs to be sent to Datadog servers.
+        /// Can be used to disable sending logs in development.
+        /// See also: `printLogsToConsole(_:)`.
+        ///
+        /// `true` by default.
+        public var sendLogsToDatadog: Bool
+
+        /// Format to use when printing logs to console - either `.short` or `.json`.
+        ///
+        /// Do not print to console by default.
+        public var consoleLogFormat: ConsoleLogFormat?
+
+        /// Set the minimum log level reported to Datadog servers.
+        /// Any log with a level equal or above the threshold will be sent.
+        ///
+        /// Note: this setting doesn't impact logs printed to the console if `printLogsToConsole(_:)`
+        /// is used - all logs will be printed, no matter of their level.
+        ///
+        /// `LogLevel.debug` by default
+        public var datadogReportingThreshold: LogLevel
```

Comment:
blocker/ [Rule 7](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3027638506/Public+API+Convention+in+iOS+SDK+2.x#Rule-7%3A-Use-of-standard-Datadog-and-RUM-terminology): _“avoid platform self referencing”_ - "Datadog" is obvious in this context.

I'd say better candidates are:
* `threshold: LogLevel` - but it lacks easy implication that this option only applies to remote not console logs
* `remoteLoggingThreshold: LogLevel`

WDYT @maxep @ganeshnj @maciejburda ?

Reply:
What about `reportingThreshold` or `reportingLevelThreshold`? 
Android uses `setDatadogLogsMinPriority` which doesn't work well on iOS either. 
So I guess we should sync with @0xnm on that one!

Reply:
I like `remoteLoggingThreshold`. To me intention is more obvious. When dealing with logging, I would need to think twice to understand that `reporting` means logging to remote (aka. sending) in that case.